x-pack/legacy/plugins/siem/public/components/markdown_editor/index.tsx

```diff
@@ -0,0 +1,135 @@
+const CommonUseField = getUseField({ component: Field });
+
+const TextArea = styled(EuiTextArea)`
+  width: 100%;
+`;
+
+const Container = styled(EuiPanel)`
+  ${({ theme }) => css`
+    padding: 0;
+    background: ${theme.eui.euiColorLightestShade};
+    position: relative;
+    .euiTab {
+      padding: 10px;
+    }
+    .euiFormRow__labelWrapper {
+      position: absolute;
+      top: -${theme.eui.euiSizeL};
+    }
+    .euiFormErrorText {
+      padding: 0 ${theme.eui.euiSizeM};
+    }
+  `}
+`;
+
+const Tabs = styled(EuiTabbedContent)`
+  width: 100%;
+`;
+
+const Footer = styled(EuiFlexGroup)`
+  ${({ theme }) => css`
+    height: 41px;
+    padding: 0 ${theme.eui.euiSizeM};
+    .euiLink {
+      font-size: ${theme.eui.euiSizeM};
+    }
+  `}
+`;
+
+const MarkdownContainer = styled(EuiPanel)`
+  min-height: 150px;
+  overflow: auto;
+`;
+
+/** An input for entering a new case description  */
+export const MarkdownEditor = React.memo<{
+  fieldName: 'comment' | 'description';
+  placeholder?: string;
+  footerContentRight?: React.ReactNode;
+  formHook?: boolean;
+  initialContent: string;
+  onChange: (description: string) => void;
+}>(({ fieldName, placeholder, footerContentRight, formHook = false, initialContent, onChange }) => {
+  const [content, setContent] = useState(initialContent);
+  useEffect(() => {
+    onChange(content);
+  }, [content]);
+  const tabs = [
+    {
+      id: fieldName,
+      name: i18n.MARKDOWN,
+      content: formHook ? (
+        <CommonUseField
+          path={fieldName}
+          onChange={e => setContent(e as string)}
+          componentProps={{
+            idAria: `case${fieldName}`,
+            'data-test-subj': `case${fieldName}`,
+            spellcheck: false,
+            euiFieldProps: { placeholder: placeholder ? placeholder : '' },
+          }}
+        />
+      ) : (
+        <TextArea
+          onChange={e => {
+            setContent(e.target.value);
+          }}
+          aria-label={`case${fieldName}`}
+          fullWidth={true}
+          placeholder={placeholder ? placeholder : ''}
+          spellCheck={false}
+          value={content}
+        />
+      ),
+    },
+    {
+      id: 'preview',
+      name: i18n.PREVIEW,
+      content: (
+        <MarkdownContainer data-test-subj="markdown-container" paddingSize="s">
+          <Markdown raw={content} />
+        </MarkdownContainer>
+      ),
+    },
+  ];
+  return (
+    <Container>
+      <Tabs
+        data-test-subj={`new-${fieldName}-tabs`}
+        size="s"
+        tabs={tabs}
+        initialSelectedTab={tabs[0]}
+      />
+      <Footer alignItems="center" gutterSize="none" justifyContent="spaceBetween">
+        <EuiFlexItem grow={false}>
+          <EuiLink href="https://www.markdownguide.org/cheat-sheet/" external target="_blank">
+            {i18n.MARKDOWN_SYNTAX_HELP}
+          </EuiLink>
+        </EuiFlexItem>
+        <EuiFlexItem grow={false}>{footerContentRight && footerContentRight}</EuiFlexItem>
```

Comment:
why do we have `footerContentRight` twice, maybe one of them should be `footerContentLeft`

Reply:
correct me if im wrong... but im checking if it exists and then if it does displaying it?